### DISCUSSION TITLE
Local development

### DISCUSSION THREAD

**Discussion by tudorvalentine** [open-sdg/open-sdg#2075] [Q&A] [answered]
Hello, 
I created 2 repositories : [sdg-site](https://github.com/tudorvalentine/sdg-site) and [sdg-data](https://github.com/tudorvalentine/sdg-data) according to the https://open-sdg.readthedocs.io/en/latest/quick-start/ . I want to launch a website locally, but I can’t do it following the steps described in https://open-sdg.readthedocs.io/en/latest/development/. I obtain :
![image](https://github.com/open-sdg/open-sdg/assets/66331277/166fda06-b6eb-4967-85be-5b78a8077d32)

But when I go to http://127.0.0.1:4000/ I get this:

![image](https://github.com/open-sdg/open-sdg/assets/66331277/9333d066-88ce-4fd8-8b03-03af714daad2)

Maybe someone encountered the same problem? thanks in advance

**Comment by brockfanning** [comment]
@tudorvalentine There may be other things going on, like with operating system issues or Ruby versions, but I have noticed recently that I need to to add a `baseurl` setting in the `_config.yml` file. In your case, try adding this:

```
baseurl: /sdg-site
```

Not sure if that will fix things but it's worth a try.

**Reply by tudorvalentine** [reply]
Ruby version is 2.6. OS is Windows 11 Pro. I added in _config.yml file
`baseurl: /sdg-site` . Also I added `remote_data_prefix: "https://tudorvalentine.github.io/sdg-data/"` because the error was appearing

![image](https://github.com/open-sdg/open-sdg/assets/66331277/3e0fafa7-21e6-4ac0-a589-655217d9a491)

**But as soon as the remote_data_prefix line was added, I got another error.**

![image](https://github.com/open-sdg/open-sdg/assets/66331277/d4be6876-cb80-4527-93aa-96e5801d2fc1)


If I try to run using the site_config.yml file downloaded from the site where `baseurl: '/sdg-site' `has been added, I get the same situation.

![image](https://github.com/open-sdg/open-sdg/assets/66331277/e228984b-6467-4e93-bfb2-2ee8953b8f8d)

**Comment by brockfanning** [comment, marked as answer]
@tudorvalentine I'm feeling like it might be an OS-level problem. I've tried in both Ubuntu Linux, as well as Windows running Ubuntu Linux inside of WSL2. FYI, I just tried it on my Ubuntu laptop and I didn't even have to change any files. For me your site repo works with just a "bundle install" and a "bundle exec jekyll serve":

```
brock@t480linux:~/repos/temp$ git clone git@github.com:tudorvalentine/sdg-site.git
Cloning into 'sdg-site'...
Warning: Permanently added the ECDSA host key for IP address '140.82.113.3' to the list of known hosts.
remote: Enumerating objects: 45101, done.
remote: Counting objects: 100% (42838/42838), done.
remote: Compressing objects: 100% (5437/5437), done.
remote: Total 45101 (delta 22310), reused 42469 (delta 21975), pack-reused 2263
Receiving objects: 100% (45101/45101), 9.86 MiB | 13.19 MiB/s, done.
Resolving deltas: 100% (22483/22483), done.
brock@t480linux:~/repos/temp$ cd sdg-site/
brock@t480linux:~/repos/temp/sdg-site$ ls
Gemfile  README.md    _config_prod.yml  _includes  _pages  _sass   scripts
LICENSE  _config.yml  _data             _layouts   _posts  assets  site_config.yml
brock@t480linux:~/repos/temp/sdg-site$ bundle install
Fetching gem metadata from https://rubygems.org/..........
Resolving dependencies...
Using public_suffix 5.0.4
Using colorator 1.1.0
Using deep_merge 1.2.2
Using eventmachine 1.2.7
Using bundler 2.2.30
Using ffi 1.16.3
Using concurrent-ruby 1.2.2
Using regexp_parser 2.8.2
Using http_parser.rb 0.8.0
Using forwardable-extended 2.6.0
Using racc 1.7.3
Using rainbow 3.1.1
Using hana 1.3.7
Using parallel 1.23.0
Using safe_yaml 1.0.5
Using unf_ext 0.0.9.1
Using rubyzip 2.3.2
Using pathutil 0.16.2
Using ethon 0.16.0
Using nokogiri 1.13.10 (x86_64-linux)
Using rb-inotify 0.10.1
Using yell 2.2.2
Using em-websocket 0.5.3
Using ecma-re-validator 0.4.0
Using rouge 3.30.0
Using liquid 4.0.4
Using mercenary 0.3.6
Using rb-fsevent 0.11.2
Using addressable 2.8.5
Using uri_template 0.7.0
Using listen 3.8.0
Using typhoeus 1.4.1
Using i18n 0.9.5
Using jekyll-watch 2.2.1
Using kramdown 1.17.0
Using sass-listen 4.0.0
Using unf 0.1.4
Using sass 3.7.4
Using html-proofer 3.19.4
Using jekyll-sass-converter 1.5.2
Using simpleidn 0.2.1
Using jekyll 3.8.4
Using json_schemer 0.2.25
Using jekyll-remote-theme 0.4.3
Using jekyll-open-sdg-plugins 2.2.0
Bundle complete! 5 Gemfile dependencies, 45 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
1 installed gem you directly depend on is looking for funding.
  Run `bundle fund` for details
brock@t480linux:~/repos/temp/sdg-site$ bundle exec jekyll serve
Configuration file: /home/brock/repos/temp/sdg-site/_config.yml
            Source: /home/brock/repos/temp/sdg-site
       Destination: /home/brock/repos/temp/sdg-site/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
Invalid theme folder: _sass
      Remote Theme: Using theme open-sdg/open-sdg
                    done in 61.388 seconds.
 Auto-regeneration: enabled for '/home/brock/repos/temp/sdg-site'
    Server address: http://127.0.0.1:4000/sdg-site/
  Server running... press ctrl-c to stop.
``` 

That results in this:
![Screenshot from 2023-11-29 22-48-39](https://github.com/open-sdg/open-sdg/assets/1319083/31a021b4-eb19-4b9d-ba8d-ce79cb15ac14)

I don't know the reason why your local site is loading the folder index instead of the index.html homepage. I don't know of any reason why working in Windows directly would not work, but I don't have any experience with that, so I can't be sure.

**Reply by tudorvalentine** [reply]
Thank you very much. It was a mistake for me that I didn't try on Ubuntu. It runs without problems on the Ubuntu operating system.
